In [9]:
import pandas as pd


In [10]:
import os
os.environ['KAGGLE_CONFIG_DIR']='/content/drive/MyDrive/minor project'

In [11]:
%cd /content/drive/MyDrive/minor project

/content/drive/MyDrive/minor project


In [12]:
!kaggle datasets download -d kmldas/loan-default-prediction

loan-default-prediction.zip: Skipping, found more recently modified local copy (use --force to force download)


In [14]:
!unzip \*.zip && rm*  .zip

Archive:  loan-default-prediction.zip
replace Default_Fin.csv? [y]es, [n]o, [A]ll, [N]one, [r]ename: y 
  inflating: Default_Fin.csv         
/bin/bash: rm*: command not found


In [16]:
df=pd.read_csv('/content/drive/MyDrive/minor project/Default_Fin.csv')

In [17]:
df.head()

Index  Employed  Bank Balance  Annual Salary  Defaulted?
0      1         1       8754.36      532339.56           0
1      2         0       9806.16      145273.56           0
2      3         1      12882.60      381205.68           0
3      4         1       6351.00      428453.88           0
4      5         1       9427.92      461562.00           0

In [18]:
df.shape

(10000, 5)

In [19]:
df.columns

Index(['Index', 'Employed', 'Bank Balance', 'Annual Salary', 'Defaulted?'], dtype='object')

**QUESTION 1:**What is the number of people who will be given a grace period of exactly 15 days?


In [20]:
count=(df[(df['Annual Salary']>100000) & (df['Bank Balance']>5000)])
print('Number of people are:',count['Index'].count())

Number of people are: 7837


**ANSWER 1:**There are 7837 people who will get a grace period of 15 days.

**QUESTION 2:**Find the maximum fine that can be levied by the bank in the period of 15 days.

In [29]:
search1=(df[((df['Annual Salary']<100000) | (df['Bank Balance']<5000)) & (df['Employed']==1)])
present_balance=search1['Bank Balance'].max()
fine=0
for j in range(0,15):
  fine=fine+present_balance*2.5/100
  present_balance=present_balance*(100-2.5)/100
print(fine)

1993.968504989993


**ANSWER 2:**The maximum fine levied by bank is Rs.1993.968

**QUESTION 3:**Assuming that all the people falling into the category of defaulters have not paid their EMI in the grace period,what is the total amount being generated by the bank as fine?
